apps/remix-ide/src/blockchain/blockchain.js

```diff
@@ -173,12 +175,33 @@ export class Blockchain extends Plugin {
   
         return this.call('terminal', 'logHtml', log)
       }
+      if (networkInfo.name === 'VM') this.config.set('vm/proxy', address)
```

Comment:
the use case is a bit limited, mainly because it won't work when deploying more than 1 instance. we can improve that later on.
As I see we could create a JSON file insted of putting it in the localstorage.
The json file would save stuff like the deployment time, the contract name, the network, ...
Then the upgrade proxy would display a dropdown of just an input when the user would need to paste the correct address from the JSON file.
can be addressed separately of course